### PR TITLE
LPS-162830 Support automatic registration of ConfigurationBeanDeclaration services

### PR DESCRIPTION
**Pull request by drewbrokke**
This is an update for [LPS-162830](https://issues.liferay.com/browse/LPS-162830).

@izaera and @rotty3000 would you be willing to look this one over and let me know if you see any issues?

/cc @pei-jung

### CODE REVIEW

File: modules/apps/portal-configuration/portal-configuration-settings/src/main/java/com/liferay/portal/configuration/settings/internal/settings/definition/ConfigurationBeanDeclarationRegistrar.java [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+					emitter.emit(service.getConfigurationBeanClass());
+
+					_bundleContext.ungetService(serviceReference);
+				});
+
+		for (Bundle bundle : bundleContext.getBundles()) {
+			_registerConfigurationBeanDeclarations(bundle);
+		}
+
+		_bundleListener = new BundleListenerImpl();
```
Comment (rotty3000) [inline modules/apps/portal-configuration/portal-configuration-settings/src/main/java/com/liferay/portal/configuration/settings/internal/settings/definition/ConfigurationBeanDeclarationRegistrar.java:77]: I would suggest changing from BundleListener to `org.osgi.util.tracker.BundleTracker`. It's easier to use following the semantics of ServiceTracker(Customizer) and covers the "retroactive check" use case in a safer way.

**Review by rotty3000 — CHANGES_REQUESTED** [review]

**Comment by melodywu** [comment]
ci:test:relevant

File: modules/apps/portal-configuration/portal-configuration-settings/src/main/java/com/liferay/portal/configuration/settings/internal/settings/definition/ConfigurationBeanDeclarationRegistrar.java (hunk 1 above)
Comment (drewbrokke) [reply modules/apps/portal-configuration/portal-configuration-settings/src/main/java/com/liferay/portal/configuration/settings/internal/settings/definition/ConfigurationBeanDeclarationRegistrar.java]: Ah, that's really nice, thank you for the suggestion @rotty3000!

**Comment by drewbrokke** [comment]
ci:test:sf

**Comment by drewbrokke** [comment]
ci:test:relevant

**Comment by drewbrokke** [comment]
ci:test:sf

**Comment by drewbrokke** [comment]
ci:test:relevant

**Comment by drewbrokke** [comment]
@patriciajeanperez is this failure real? Would it be worth re-running `relevant`?

**Comment by drewbrokke** [comment]
ci:test:relevant

**Comment by drewbrokke** [comment]
ci:test:relevant

**Comment by epark18** [comment]
Hi @drewbrokke. I believe your changes to the _SettingsLocatorHelperImpl.java_ file is the cause of the relevant failures according to the job summary. It looks like the changes are causing the tests to run longer and the build is timing out. 

There was an issue with loggers in Brian's upstream recently. Details are in this [thread](https://liferay.slack.com/archives/CLCD3DQLF/p1663051635929259). Shuyang mentioned a problem with logs so maybe this could be helpful.

**Comment by drewbrokke** [comment]
ci:reevaluate:1287401_4423

**Comment by drewbrokke** [comment]
ci:reevaluate:1287401_4423

**Comment by drewbrokke** [comment]
ci:test:relevant

**Comment by drewbrokke** [comment]
ci:test:relevant

**Comment by drewbrokke** [comment]
ci:test:sf

**Comment by drewbrokke** [comment]
ci:test:relevant

**Comment by drewbrokke** [comment]
ci:test:sf

**Comment by Sharryxue** [comment]
ci:test:system

**Comment by Sharryxue** [comment]
No related issue found in ci:test:system.

**Comment by drewbrokke** [comment]
Superseded by https://github.com/dantewang/liferay-portal/pull/2269

**Pull request closed without merging**